crates/ruff_linter/src/rules/flake8_bandit/rules/snmp_insecure_version.rs

```diff
@@ -47,10 +47,11 @@ pub(crate) fn snmp_insecure_version(checker: &Checker, call: &ast::ExprCall) {
         .semantic()
         .resolve_qualified_name(&call.func)
         .is_some_and(|qualified_name| {
-            matches!(
-                qualified_name.segments(),
-                ["pysnmp", "hlapi", "CommunityData"]
-            )
+            let segments = qualified_name.segments();
+            segments.len() >= 3
+                && segments[0] == "pysnmp"
+                && segments[1] == "hlapi"
+                && segments[segments.len() - 1] == "CommunityData"
```

Comment:
I think we can still use a `matches` here:

```suggestion
           matches!(qualified_name.segments(),
            [
                "pysnmp",
                 "hlapi", ..,
                 "CommunityData"
                 ]
                 )
```

but it would be better still to use `|` to specify the two options in the `..` position because `..` will allow any number of entries, and I think there's only one alternative right?

Unfortunately we might have to do `["pysnmp", "hlapi", "CommunityData"] | ["pysnmp", "hlapi", "auth", "CommunityData"]`, but I still think that's more readable than the indexing.